File: books-practice/Data Structures and  Algorithms in Java/src/chap_3/doubleLL/ds/java/Node.java
package chap_3.doubleLL.ds.java;
public class Node<E> {
  private E element;
  private Node<E> prev;
  private Node<E> next;

  public Node(E e, Node<E> p, Node<E> n) {
    element = e;
    prev = p;
    next = n;
  }

  public E getElement() {
    return this.element;
  }

  public Node<E> getPrev() {
    return this.prev;
  }

  public Node<E> getNext() {
    return this.next;
  }

  public void setPrev(Node<E> p) {
    this.prev = p;
  }

  public void setNext(Node<E> n) {
    this.next = n;
  }
}

File: books-practice/Data Structures and  Algorithms in Java/src/chap_3/circular_linked_list/ds/javascript/node.ts
class NodeCLL<T> {
  private element: T;
  private next: NodeCLL<T> | null;

  constructor(e: T, n: NodeCLL<T> | null) {
    this.element = e;
    this.next = n;
  }

  public getElement(): T {
    return this.element;
  }

  public getNext(): NodeCLL<T> | null {
    return this.next;
  }

  public setNext(n: NodeCLL<T>) {
    this.next = n;
  }
}

export default NodeCLL;